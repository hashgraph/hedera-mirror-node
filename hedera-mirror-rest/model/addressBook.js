/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

class AddressBook {
  /**
   * Parses address_book table columns into object
   */
  constructor(addressBook) {
    // explicitly assign properties to restict properties and allow for composition in other models
    this.endConsensusTimestamp = addressBook.end_consensus_timestamp;
    this.fileData = addressBook.file_data;
    this.fileId = addressBook.file_id;
    this.nodeCount = addressBook.node_count;
    this.startConsensusTimestamp = addressBook.start_consensus_timestamp;
  }

  static tableAlias = 'adb';
  static tableName = 'address_book';

  static END_CONSENSUS_TIMESTAMP = 'end_consensus_timestamp';
  static FILE_DATA = 'file_data';
  static NODE_COUNT = 'node_count';
  static FILE_ID = 'file_id';
  static START_CONSENSUS_TIMESTAMP = 'start_consensus_timestamp';

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default AddressBook;
