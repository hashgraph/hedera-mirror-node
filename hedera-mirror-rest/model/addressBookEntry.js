/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

class AddressBookEntry {
  /**
   * Parses address_book_entry table columns into object
   */
  constructor(addressBookEntry) {
    // explicitly assign properties to restrict properties and allow for composition in other models
    this.consensusTimestamp = addressBookEntry.consensus_timestamp;
    this.description = addressBookEntry.description;
    this.memo = addressBookEntry.memo;
    this.nodeAccountId = addressBookEntry.node_account_id;
    this.nodeCertHash = addressBookEntry.node_cert_hash;
    this.nodeId = addressBookEntry.node_id;
    this.publicKey = addressBookEntry.public_key;
    this.stake = addressBookEntry.stake;
  }

  static tableAlias = 'abe';
  static tableName = 'address_book_entry';

  static CONSENSUS_TIMESTAMP = 'consensus_timestamp';
  static MEMO = 'memo';
  static PUBLIC_KEY = 'public_key';
  static NODE_ID = 'node_id';
  static NODE_ACCOUNT_ID = 'node_account_id';
  static NODE_CERT_HASH = 'node_cert_hash';
  static DESCRIPTION = 'description';
  static STAKE = 'stake';

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default AddressBookEntry;
