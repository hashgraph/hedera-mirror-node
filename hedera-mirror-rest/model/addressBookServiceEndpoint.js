/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

class AddressBookServiceEndpoint {
  /**
   * Parses crypto_allowance table columns into object
   */
  constructor(serviceEndpoint) {
    // explicitly assign properties to restict properties and allow for composition in other models
    this.consensusTimestamp = serviceEndpoint.consensus_timestamp;
    this.ipAddressV4 = serviceEndpoint.ip_address_v4;
    this.nodeId = serviceEndpoint.node_id;
    this.port = serviceEndpoint.port;
  }

  static tableAlias = 'abse';
  static tableName = 'address_book_service_endpoint';

  static CONSENSUS_TIMESTAMP = 'consensus_timestamp';
  static IP_ADDRESS_V4 = 'ip_address_v4';
  static NODE_ID = 'node_id';
  static PORT = 'port';

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default AddressBookServiceEndpoint;
