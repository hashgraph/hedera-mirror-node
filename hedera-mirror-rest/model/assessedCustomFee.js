/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

class AssessedCustomFee {
  /**
   * Parses assessed_custom_fee table columns into object
   */
  constructor(assessedCustomFee) {
    this.amount = assessedCustomFee.amount;
    this.collectorAccountId = assessedCustomFee.collector_account_id;
    this.consensusTimestamp = assessedCustomFee.consensus_timestamp;
    this.effectivePayerAccountIds = assessedCustomFee.effective_payer_account_ids;
    this.tokenId = assessedCustomFee.token_id;
  }

  static tableAlias = 'acf';
  static tableName = 'assessed_custom_fee';

  static AMOUNT = `amount`;
  static COLLECTOR_ACCOUNT_ID = `collector_account_id`;
  static CONSENSUS_TIMESTAMP = `consensus_timestamp`;
  static EFFECTIVE_PAYER_ACCOUNT_IDS = `effective_payer_account_ids`;
  static PAYER_ACCOUNT_ID = 'payer_account_id';
  static TOKEN_ID = `token_id`;

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default AssessedCustomFee;
