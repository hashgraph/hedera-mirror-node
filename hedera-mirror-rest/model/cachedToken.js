/*
 * Copyright (C) 2024 Hedera Hashgraph, LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import TokenFreezeStatus from './tokenFreezeStatus';
import TokenKycStatus from './tokenKycStatus';

const tokenFreezeValues = TokenFreezeStatus.VALUES;
const tokenKycValues = TokenKycStatus.VALUES;

/**
 * Cached token object to store token's decimals, freeze default and kyc default. Note freeze_default is not the same as
 * the freeze_default column in the token table. freeze_default / kyc_default is either copied from the SQL query result
 * directly or determined based on the freeze_default & freeze_key / kyc_key columns in the token table.
 */
class CachedToken {
  constructor(token) {
    this.decimals = token.decimals;

    if (token.freeze_key !== undefined) {
      if (token.freeze_key !== null) {
        this.freezeDefault = token.freeze_default ? tokenFreezeValues['FROZEN'] : tokenFreezeValues['UNFROZEN'];
      } else {
        this.freezeDefault = tokenFreezeValues['NOT_APPLICABLE'];
      }
    } else {
      this.freezeDefault = token.freeze_default;
    }

    if (token.kyc_key !== undefined) {
      this.kycDefault = token.kyc_key ? tokenKycValues['REVOKED'] : tokenKycValues['NOT_APPLICABLE'];
    } else {
      this.kycDefault = token.kyc_default;
    }

    this.tokenId = token.token_id;
  }
}

export default CachedToken;
