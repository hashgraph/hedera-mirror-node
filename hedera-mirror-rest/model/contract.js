/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

import _ from 'lodash';

class Contract {
  static tableAlias = 'c';
  static tableName = 'contract';

  static FILE_ID = 'file_id';
  static ID = 'id';
  static INITCODE = 'initcode';
  static RUNTIME_BYTECODE = 'runtime_bytecode';

  /**
   * Parses contract table columns into object
   */
  constructor(contract) {
    Object.assign(
      this,
      _.mapKeys(contract, (v, k) => _.camelCase(k))
    );
  }

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default Contract;
