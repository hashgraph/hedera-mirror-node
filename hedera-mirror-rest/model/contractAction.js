/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

import _ from 'lodash';

class ContractAction {
  static tableAlias = 'cact';
  static tableName = 'contract_action';

  static CALL_DEPTH = 'call_depth';
  static CALL_OPERATION_TYPE = 'call_operation_type';
  static CALL_TYPE = 'call_type';
  static CALLER = 'caller';
  static CALLER_TYPE = 'caller_type';
  static CONSENSUS_TIMESTAMP = 'consensus_timestamp';
  static GAS = 'gas';
  static GAS_USED = 'gas_used';
  static INDEX = 'index';
  static INPUT = 'input';
  static PAYER_ACCOUNT_ID = 'payer_account_id';
  static RECIPIENT_ACCOUNT = 'recipient_account';
  static RECIPIENT_ADDRESS = 'recipient_address';
  static RECIPIENT_CONTRACT = 'recipient_contract';
  static RESULT_DATA = 'result_data';
  static RESULT_DATA_TYPE = 'result_data_type';
  static VALUE = 'value';

  /**
   * Parses contract_action table columns into object
   */
  constructor(contractAction) {
    Object.assign(
      this,
      _.mapKeys(contractAction, (v, k) => _.camelCase(k))
    );
  }

  /**
   * Gets full column name with table alias prepended.
   *
   * @param {string} columnName
   * @private
   */
  static getFullName(columnName) {
    return `${this.tableAlias}.${columnName}`;
  }
}

export default ContractAction;
