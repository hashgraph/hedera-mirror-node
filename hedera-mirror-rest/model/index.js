/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

import AccountBalanceFile from './accountBalanceFile';
import AddressBook from './addressBook';
import AddressBookEntry from './addressBookEntry';
import AddressBookServiceEndpoint from './addressBookServiceEndpoint';
import AssessedCustomFee from './assessedCustomFee';
import Contract from './contract';
import CryptoAllowance from './cryptoAllowance';
import ContractAction from './contractAction.js';
import ContractLog from './contractLog';
import ContractResult from './contractResult';
import ContractStateChange from './contractStateChange';
import CryptoTransfer from './cryptoTransfer';
import CustomFee from './customFee';
import Entity from './entity';
import EthereumTransaction from './ethereumTransaction';
import ExchangeRate from './exchangeRate';
import FileData from './fileData';
import NetworkNode from './networkNode';
import NetworkStake from './networkStake.js';
import Nft from './nft';
import NftTransfer from './nftTransfer';
import NodeStake from './nodeStake';
import RecordFile from './recordFile';
import SignatureType from './signatureType';
import StakingRewardTransfer from './stakingRewardTransfer';
import Token from './token';
import TokenAllowance from './tokenAllowance';
import TokenFreezeStatus from './tokenFreezeStatus';
import TokenKycStatus from './tokenKycStatus';
import TokenAccount from './tokenAccount';
import TokenBalance from './tokenBalance';
import TokenTransfer from './tokenTransfer';
import TokenType from './tokenType';
import TopicMessage from './topicMessage';
import Transaction from './transaction';
import TransactionHash from './transactionHash';
import TransactionId from './transactionId';
import TransactionResult from './transactionResult';
import TransactionType from './transactionType';
import FeeSchedule from './feeSchedule';
import ContractState from './contractState';

export {
  AccountBalanceFile,
  AddressBook,
  AddressBookEntry,
  AddressBookServiceEndpoint,
  AssessedCustomFee,
  Contract,
  CryptoAllowance,
  ContractAction,
  ContractLog,
  ContractResult,
  ContractStateChange,
  CryptoTransfer,
  CustomFee,
  Entity,
  EthereumTransaction,
  ExchangeRate,
  FileData,
  NetworkNode,
  NetworkStake,
  Nft,
  NftTransfer,
  NodeStake,
  RecordFile,
  SignatureType,
  StakingRewardTransfer,
  Token,
  TokenAllowance,
  TokenFreezeStatus,
  TokenKycStatus,
  TokenAccount,
  TokenBalance,
  TokenTransfer,
  TokenType,
  TopicMessage,
  Transaction,
  TransactionHash,
  TransactionId,
  TransactionResult,
  TransactionType,
  FeeSchedule,
  ContractState,
};
