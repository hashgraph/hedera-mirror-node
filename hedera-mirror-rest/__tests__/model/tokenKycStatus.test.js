/*-
 * ‌
 * Hedera Mirror Node
 * ​
 * Copyright (C) 2019 - 2023 Hedera Hashgraph, LLC
 * ​
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ‍
 */

import {TokenKycStatus} from '../../model';

describe('TokenKycStatus', () => {
  describe('invalid id', () => {
    [-1, 3].forEach((value) =>
      test(`${value}`, () => {
        expect(() => new TokenKycStatus(value)).toThrowErrorMatchingSnapshot();
      })
    );
  });

  test('toJSON', () => {
    const input = [new TokenKycStatus(0), new TokenKycStatus(1), new TokenKycStatus(2)];
    expect(JSON.stringify(input)).toEqual(JSON.stringify(['NOT_APPLICABLE', 'GRANTED', 'REVOKED']));
  });
});
