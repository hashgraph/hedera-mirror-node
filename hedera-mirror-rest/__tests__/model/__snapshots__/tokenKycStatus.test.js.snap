// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenKycStatus invalid id -1 1`] = `"Invalid token kyc status id -1"`;

exports[`TokenKycStatus invalid id 3 1`] = `"Invalid token kyc status id 3"`;
